run_infrastructure.py

```diff
@@ -42,23 +42,23 @@ def run_infrastructure(
     """
 
     if len(datastores) == 0:
-        os.system(
+        _run_cmd_or_err(
             f'echo "no datastores specified, configuring infrastructure for all datastores"'
         )
         datastores = DOCKERFILE_DATASTORES + EXTERNAL_DATASTORES
     else:
-        os.system(f'echo "datastores specified: {", ".join(datastores)}"')
+        _run_cmd_or_err(f'echo "datastores specified: {", ".join(datastores)}"')
 
     # De-duplicate datastores
     datastores = set(datastores)
 
     # Configure docker-compose path
     path: str = get_path_for_datastores(datastores)
 
-    os.system(f'echo "infrastructure path: {path}"')
-    os.system(f"docker-compose {path} build")
-    os.system(f'echo "sleeping for: {DOCKER_WAIT} while infrastructure loads"')
-    os.system(f"sleep {DOCKER_WAIT}")
+    _run_cmd_or_err(f'echo "infrastructure path: {path}"')
+    _run_cmd_or_err(f"docker-compose {path} up -d")
```

Comment:
Discussed in person - this is for the mssql script - we were trying to run the mssql script that populates the database but the container hadn't come up yet, this forces us to wait for the containers to come up and for the wait to be respected, so we avoid those mssql failures we were getting.

Reply:
To clarify — `docker-compose {path} build` is akin to a lazyload of the infra. The containers get built in the background but not run until they're forced to with some `docker-compose run` or `docker-compose exec` command. This meant that the call to run `../mssql_setup.py` that's invoked in `seed_initial_data()` below would routinely fail, as the container wasn't up by the time the script ran.